mixer/adapter/cloudwatch/logHandler_test.go

```diff
@@ -0,0 +1,337 @@
+}
+
+func (m *mockFailCloudWatchLogsClient) PutLogEvents(input *cloudwatchlogs.PutLogEventsInput) (*cloudwatchlogs.PutLogEventsOutput, error) {
+	return nil, errors.New("put logentry data failed")
+}
+
+func (m *mockFailCloudWatchLogsClient) DescribeLogStreams(input *cloudwatchlogs.DescribeLogStreamsInput) (*cloudwatchlogs.DescribeLogStreamsOutput, error) {
+	return m.resp, m.describeError
+}
+
+func generateLogStreamOutput() *cloudwatchlogs.DescribeLogStreamsOutput {
+	logstreams := make([]*cloudwatchlogs.LogStream, 0, 1)
+
+	stream := cloudwatchlogs.LogStream{
+		LogStreamName:       aws.String("TestLogStream"),
+		UploadSequenceToken: aws.String("49579643037721145729486712515534281748446571659784111634"),
+	}
+
+	logstreams = append(logstreams, &stream)
+
+	return &cloudwatchlogs.DescribeLogStreamsOutput{
+		LogStreams: logstreams,
+	}
+}
+
+func generateEmptyLogStreamOutput() *cloudwatchlogs.DescribeLogStreamsOutput {
+	logstreams := make([]*cloudwatchlogs.LogStream, 0, 0)
+	return &cloudwatchlogs.DescribeLogStreamsOutput{
+		LogStreams: logstreams,
+	}
+}
+
+func generateNilLogStreamOutput() *cloudwatchlogs.DescribeLogStreamsOutput {
+	return &cloudwatchlogs.DescribeLogStreamsOutput{
+		LogStreams: nil,
+	}
+}
+
+func TestPutLogEntryData(t *testing.T) {
+	cfg := &config.Params{
+		LogGroupName:  "TestLogGroup",
+		LogStreamName: "TestLogStream",
+	}
+
+	env := test.NewEnv(t)
+
+	cases := []struct {
+		logEntryData      []*cloudwatchlogs.InputLogEvent
+		expectedErrString string
+		handler           adapter.Handler
+		nextSequenceToken string
+	}{
+		{
+			[]*cloudwatchlogs.InputLogEvent{
+				{Message: aws.String("testMessage2"), Timestamp: aws.Int64(1)},
+			},
+			"",
+			newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockCloudWatchLogsClient{}),
+			"49579643037721145729486712515534281748446571659784111634",
+		},
+		{
+			[]*cloudwatchlogs.InputLogEvent{},
+			"put logentry data",
+			newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockFailCloudWatchLogsClient{}),
+			"49579643037721145729486712515534281748446571659784111634",
+		},
+		{
+			[]*cloudwatchlogs.InputLogEvent{
+				{Message: aws.String("testMessage2"), Timestamp: aws.Int64(1)},
+			},
+			"put logentry data",
+			newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockFailCloudWatchLogsClient{}),
+			"49579643037721145729486712515534281748446571659784111634",
+		},
+		{
+			[]*cloudwatchlogs.InputLogEvent{
+				{Message: aws.String("testMessage2"), Timestamp: aws.Int64(1)},
+			},
+			"put logentry data",
+			newHandler(nil, nil, env, cfg, &mockCloudWatchClient{}, &mockFailCloudWatchLogsClient{}),
+			"",
+		},
+	}
+
+	for _, c := range cases {
+		h, ok := c.handler.(*handler)
+		if !ok {
+			t.Error("test case has the wrong type of handler.")
+		}
+
+		err := h.putLogEntryData(c.logEntryData, c.nextSequenceToken)
+		if len(c.expectedErrString) > 0 {
+			if err == nil {
+				t.Errorf("expected an error containing %s, got none", c.expectedErrString)
```

Comment:
super nit: `t.Errorf("putLogEntryData() did not produce expected error: %s", c.expectedErrString)`

Reply:
done